Userland/Libraries/LibWeb/WebSockets/WebSocket.cpp

```diff
@@ -50,50 +51,107 @@ WebSocketClientManager::WebSocketClientManager() = default;
 // https://websockets.spec.whatwg.org/#dom-websocket-websocket
 WebIDL::ExceptionOr<JS::NonnullGCPtr<WebSocket>> WebSocket::construct_impl(JS::Realm& realm, String const& url, Optional<Variant<String, Vector<String>>> const& protocols)
 {
-    auto& window = verify_cast<HTML::Window>(realm.global_object());
-    AK::URL url_record(url);
+    auto& vm = realm.vm();
+
+    auto web_socket = MUST_OR_THROW_OOM(realm.heap().allocate<WebSocket>(realm, realm));
+    auto& relevant_settings_object = HTML::relevant_settings_object(*web_socket);
+
+    // 1. Let baseURL be this's relevant settings object's API base URL.
+    auto base_url = relevant_settings_object.api_base_url();
+
+    // 2. Let urlRecord be the result of applying the URL parser to url with baseURL.
+    auto url_record = base_url.complete_url(url);
```

Comment:
Maybe I am misunderstanding, but I think this is calling https://url.spec.whatwg.org/#concept-basic-url-parser rather than https://url.spec.whatwg.org/#concept-url-parser - seems to be calling `URL URLParser::parse` which links to the 'basic' parse.

Reply:
Great catch, I have left it as a FIXME for now